src/ring_theory/ideal/nc_jacobson_radical.lean

```diff
@@ -0,0 +1,251 @@
+end
+
+lemma has_left_inv_one_add_of_mem_jacobson {x : R} :
+  x ∈ nc_jacobson R → has_left_inv (1 + x):=
+begin
+  contrapose,
+  rw not_has_left_inv_iff_mem_maximal,
+  rintro ⟨I, hImax, hxxI⟩ hx,
+  refine hImax.ne_top _,
+  rw eq_top_iff_one,
+  have hxI : x ∈ I := by { rw [nc_jacobson, mem_Inf] at hx, apply hx hImax },
+  exact (add_mem_cancel_right hxI).mp hxxI,
+end
+
+lemma one_add_mul_self_mem_maximal_of_not_mem_maximal {x : R} {I : ideal R}
+  (h₁ : I.is_maximal) (h₂ : x ∉ I) : ∃ a : R, 1 + a * x ∈ I :=
+begin
+  have : (1 : R) ∈ I ⊔ span {x},
+  { rw is_maximal_iff at h₁,
+    apply h₁.2 _ _ le_sup_left h₂,
+    apply mem_sup_right,
+    apply submodule.mem_span_singleton_self },
+  rw submodule.mem_sup at this,
+  obtain ⟨m, hmI, y, hy, hmy⟩ := this,
+  rw mem_span_singleton' at hy,
+  obtain ⟨a, rfl⟩ := hy,
+  existsi -a,
+  rwa [←hmy, neg_mul, add_neg_cancel_right],
+end
+
+/-- Characterizations of the Jacobson radical of a ring.
+
+The following are equivalent for an element `x` in a ring `R`.
+* 0: `x` is in the Jacobson radical of `R`, that is, contained in every mximal left ideal.
+* 1: `1 + a * x` has a left inverse for any `a : R`.
+* 2: `1 + a * x` is a unit for any `a : R`.
+* 3: `1 + x * b` is a unit for any `b : R`.
+* 4: `1 + a * x * b` is a unit for any `a b : R`.
+-/
+theorem mem_nc_jacobson_tfae {R : Type u} [ring R] (x : R) : tfae [
+  x ∈ nc_jacobson R,
+  ∀ a : R, has_left_inv (1 + a * x),
+  ∀ a : R, is_unit (1 + a * x),
+  ∀ b : R, is_unit (1 + x * b),
+  ∀ a b : R, is_unit (1 + a * x * b)] :=
+begin
+  tfae_have : 1 → 2,
+  { exact λ hx a, has_left_inv_one_add_of_mem_jacobson $
+    (nc_jacobson R).smul_mem' a hx },
+  tfae_have : 2 → 3,
+  { intros hx a,
+    obtain ⟨c, hc⟩ := hx a,
+    apply is_unit_of_has_left_inv_of_has_left_inv hc,
+    suffices : c = 1 + ( -c * a * x),
+    { rw this, apply hx },
+    { calc c = c * (1 + a * x) + ( -c * a * x) : by noncomm_ring
+        ...  = 1 + ( -c * a * x) : by rw hc } },
+  tfae_have : 3 → 5,
+  { intros hx _ _,
+    apply is_unit.one_add_mul_swap,
+    rw ←mul_assoc,
+    apply hx },
+  tfae_have : 5 → 1,
+  { intro h,
+    by_contra hx,
+    rw [nc_jacobson, submodule.mem_Inf] at hx,
+    simp only [not_forall] at hx,
+    rcases hx with ⟨I, hImax, hxI⟩,
+    refine hImax.ne_top _,
+    obtain ⟨a, ha⟩ := one_add_mul_self_mem_maximal_of_not_mem_maximal hImax hxI,
+    apply eq_top_of_is_unit_mem _ ha,
+    specialize h a 1,
+    rwa [mul_assoc, mul_one] at h },
+  tfae_have : 3 ↔ 4,
+  { split; exact λ h b, (h b).one_add_mul_swap },
+  tfae_finish,
+end
+
+/--
+The definition of the Jacobson radical of a ring is left-right symmetric, that is,
+the intersection of all maximal left ideals coincides with that of all maximal right ideals.
+We express this by using the opposite ring `Rᵐᵒᵖ`, that is,
+the Jacobson radical of `R` coincides with that of `Rᵐᵒᵖ`,
+-/
+theorem nc_jacobson_symm {x : R} :
+  x ∈ nc_jacobson R ↔ op x ∈ nc_jacobson Rᵐᵒᵖ :=
```

Comment:
My understanding is that `nc_jacobson Rᵐᵒᵖ` is not a right-ideal; that would be `submodule Rᵐᵒᵖ R`, but what you have here is `submodule Rᵐᵒᵖ Rᵐᵒᵖ`.

Reply:
Yes, you're right (I haven't realized that we can consider right ideals in this way). Then maybe the direct statement of this doc-string looks like:
```
example : (nc_jacobson R : set R) = ↑ (Inf {I : submodule Rᵐᵒᵖ R | is_coatom I })
```
I think the proof of it will be a longer than `nc_jacobson_symm` because we have to compare `nc_jacobson Rᵐᵒᵖ` and RHS (and this formalization looks uglier).
Maybe is it better only to change a doc-string?

Reply:
Certainly for now you should change the docstring; and maybe add a Todo comment about the statement you thought you were making

Reply:
OK thank you, I changed the docstring, and rename it to `nc_jacobson_op`.